sdk/python/foundation-models/system/inference/text-generation/llama-files/score/hf-tgi/score.py

```diff
@@ -0,0 +1,460 @@
+
+    logger.info(f"Server process {TEXT_GEN_LAUNCHER_PROCESS_NAME} running. Hitting endpoint with 5s delay")
+    time.sleep(5)
+
+    payload_dict = {
+        "inputs": "Meaning of life is",
+        "parameters":{"max_new_tokens":2}
+    }
+
+    json_str = json.dumps(payload_dict)
+
+    try:
+        response = requests.post(
+            url=LOCAL_HOST_URI,
+            data=json_str,
+            headers={
+                "Content-Type": "application/json"
+            }
+        )
+        logger.info(f"response status code: {response.status_code}")
+        if response.status_code == 200 or response.status_code == 201:
+            return True
+    except Exception as e:
+        logger.warning(f"Test request failed. Error {e}")
+    return False
+
+
+def init():
+    global client
+    global task_type
+
+    try:
+        model_id = os.environ.get(MODEL_ID, DEFAULT_MODEL_ID_PATH)
+        client_timeout = os.environ.get(CLIENT_TIMEOUT, MAX_REQUEST_TIMEOUT)
+
+        for k, v in os.environ.items():
+            logger.info(f"env: {k} = {v}")
+
+        model_path = os.path.join(os.getenv("AZUREML_MODEL_DIR", ""), model_id)
+        model_path = os.path.join(model_path, "model")  # default model path for MLFlow model
+        # Use safetensors if present
+        if os.path.isdir(os.path.join(model_path, "safetensors")):
+            model_path = os.path.join(model_path, "safetensors")
+        
+        abs_mlmodel_path = os.path.join(os.getenv("AZUREML_MODEL_DIR", ""), MLMODEL_PATH)
+        mlmodel = {}
+        if abs_mlmodel_path and os.path.exists(abs_mlmodel_path):
+            with open(abs_mlmodel_path) as f:
+                mlmodel = yaml.safe_load(f)
+
+        if mlmodel:
+            flavors = mlmodel.get("flavors", {})
+            if "hftransformersv2" in flavors:
+                task_type = flavors["hftransformersv2"]["task_type"]
+                if task_type not in (SupportedTask.TEXT_GENERATION, SupportedTask.CHAT_COMPLETION):
+                    raise Exception(f"Unsupported task_type {task_type}")
+
+        logger.info(f"Loading model from path {model_path} for task_type: {task_type}")
+        logger.info(f"List model_path = {os.listdir(model_path)}")
+
+        logger.info("Starting server")
+        cmd = f"text-generation-launcher --model-id {model_path} &"
+        os.system(cmd)
+        time.sleep(20)
+
+        WAIT_TIME = 60
+        while not is_server_healthy():
+            logger.info(f"Server not up. Waiting for {WAIT_TIME}s, before querying again.")
+            time.sleep(WAIT_TIME)
+
+        logger.info("Server Started")
+        client = Client(LOCAL_HOST_URI, timeout=client_timeout)  # use deployment settings
+        logger.info(f"Created Client: {client}")
+    except Exception as e:
+        raise Exception(f"Error in creating client or server: {e}")
+
+
+"""
+Read about client accepted parameters here:
```

Comment:
why do we want to keep it in the score.py file?

Reply:
Done